### DISCUSSION TITLE
Is there a way to mark a property as deleted in an ecschema?

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#5415] [Q&A] [answered]
Hello I am working on the FutureOn connector to iTwin. We have an ecschema.xml which has been developed over the last few years, as the project develops, there are some properties in the schema that are no longer correct, and need to be re-defined. For example changing a property from a string to a double.

It is not possble to change the definition of an existing property, or to delete an existing property in a ecschema, becuase this will be rejected by the iModel as not compatible.

Is there a way to mark a property as no longer required? So it would still exist within the schema to maintain compatibility, but will no longer be displayed in the iModel.

**Comment by grigasp** [comment]
@ColinKerr @diegoalexdiaz

**Reply by diegoalexdiaz** [reply]
Hello @PeterJBassett, 

You are right that certain modifications of BIS schemas are considered "major upgrades" and are not allowed to be applied "in-place" as they would potentially break compatibility with applications written for the previous schema. More details on this topic can be found in this [article](https://www.itwinjs.org/bis/guide/schema-evolution/schema-versioning-and-generations/#ec-schema-versioning).

However, you can mark a property as deprecated as well as hide it from the UX by using a couple of Custom-Attributes. e.g.

```
<ECProperty propertyName="XXX" ...>
    <ECCustomAttributes>
        <HiddenProperty xmlns="CoreCustomAttributes.01.00.03"/>
        <Deprecated xmlns="CoreCustomAttributes.01.00.03">
            <Description>Any comment advising existing consumers of this property about what to do to read the same kind of information</Description>
        </Deprecated>
    </ECCustomAttributes>
</ECProperty>
```

**Reply by PeterJBassett** [reply]
Hello @diegoalexdiaz - Thank you so much for your help. This should solve my problem. 
One more question, is there a link to any documentation that describes these custom attributes?

**Reply by diegoalexdiaz** [reply, marked as answer]
Happy to help. Please refer to the following [link](https://www.itwinjs.org/bis/domains/corecustomattributes.ecschema/) for documentation on the CoreCustomAttributes schema.
